### PR TITLE
Kubevirt

### PR DESCRIPTION
**Pull request by iranzo**
Add kubevirt scenario based on minikube tutorial

### CODE REVIEW

File: kubevirt/finish.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Well done, you have completed the KubeVirt demo!
+
+Hope you've enjoyed it, for more information about Kubevirt, please do check <https://kubevirt.io>!
```
Comment (codificat) [inline kubevirt/finish.md]: KubeVirt not Kubevirt.

Also probably "We hope you enjoyed it.", period instead of comman (and capitalize).

File: kubevirt/intro.md [hunk 2]
```diff
@@ -0,0 +1 @@
+This scenario demonstrates how kubevirt can be used to start a virtual machine
```
Comment (codificat) [inline kubevirt/intro.md]: KubeVirt.

Also it's probably worth expanding the introduction a bit more. No need to get into details, but at least mention that it's about managing virtual machines within a Kubernetes cluster.

File: kubevirt/step1.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+### Wait for the cluster to be ready
+
+Before we can start, we need to wait for the Kubernetes cluster to be ready
```
Comment (codificat) [inline kubevirt/step1.md]: ... and we will know that it's ready when ____ (fill in the gap)

**Review by ptrnull — COMMENTED** [review]
/lgtm
/approve

File: kubevirt/step1.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+#### Deploy KubeVirt
+
+Deploy KubeVirt operator using latest Kubevirt Version
```
Comment (codificat) [inline kubevirt/step1.md]: Kube**V**irt **v**ersion (swap capital V).

It would probably be nicer to elaborate a bit more too... what's this operator thing? What are we actually doing here?

File: kubevirt/step1.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases|grep tag_name|sort -V | tail -1 | awk -F':' '{print $2}' | sed 's/,//' | xargs)
```
Comment (codificat) [inline kubevirt/step1.md]: That's a relatively ugly command... and it's the very first one in the scenario. Maybe we can make it a bit more friendly?

The environment does have `jq` installed, which can help.

File: kubevirt/step1.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+echo $KUBEVIRT_VERSION`{{execute}}
+
+`kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-operator.yaml`{{execute}}
+
+Provide some initial configuration
+
+`kubectl create configmap kubevirt-config -n kubevirt --from-literal debug.useEmulation=true`{{execute}}
+
+Above commands enables 'emulation' to run the VM's as our demo environment is using 'emulated' virtualization.
```
Comment (codificat) [inline kubevirt/step1.md]: Better move the explanation of what the command does before the command, i.e. merge with the "provide initial configuration" part.

Again it might help to elaborate slightly more: why are we using *'emulated' virtualization*? It might sound obvious, but maybe not...

File: kubevirt/step1.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+Now let's deploy kubevirt by creating a custom resource:
```
Comment (codificat) [inline kubevirt/step1.md]: KubeVirt

Comment (codificat) [reply kubevirt/step1.md]: And how does that "custom resource" relate to "deploy KubeVirt"?

File: kubevirt/step1.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+virt-controller-6849d45bcc-88zd4   1/1       Running   0          3m
+virt-controller-6849d45bcc-cmfzk   1/1       Running   0          3m
+virt-handler-fvsqw                 1/1       Running   0          3m
+virt-operator-5649f67475-gmphg     1/1       Running   0          4m
+virt-operator-5649f67475-sw78k     1/1       Running   0          4m
+~~~
+
+#### Install Virtctl
+
+`virtctl` is a client utility to provide some more convenient ways to interact with the VM:
```
Comment (codificat) [inline kubevirt/step1.md]: ... interact with virtual machines.

Maybe mention a couple of things it can do: start, stop, console... to help understand why we need it.

File: kubevirt/step2.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+#### Deploy a VM
+
+Start of by creating a virtual machine:
```
Comment (codificat) [inline kubevirt/step2.md]: Grammar: remove "of" - or did you mean "off"?

Could we elaborate a bit more here? What does this command exactly do?

Say, I'm used to my typical VM management environment... this sounds strange to me.

Or, say, I'm used to typical Kubernetes resource creation, but never thought I could "create a VM" with `kubectl apply`. How so?

File: kubevirt/step2.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+`kubectl apply -f https://raw.githubusercontent.com/kubevirt/demo/master/manifests/vm.yaml`{{execute}}
+
+After it has been created you can manage VMs using the usual verbs:
```
Comment (codificat) [inline kubevirt/step2.md]: Grammar: "it" (singular) vs "VMs" (plural)

Also, the "usual verbs" (which might not be "usual" to some people" do not actually let you do what one typically understands as "manage VMs", I thought from the previous step that this is the reason why we need `virtctl`, no?

File: kubevirt/step2.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+```
+$ kubectl get vms
+$ kubectl get vms -o yaml testvm
+```
+
+Check VM's defined (using above commands)
+`kubectl get vms`{{execute}}
+
+To start a VM, virtctl can be used:
```
Comment (codificat) [inline kubevirt/step2.md]: Can or should? In which cases?

File: kubevirt/step2.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+`./virtctl start testvm`{{execute}}
+
+Alternatively you can use `kubectl edit vm testvm` to set `.spec.running: true`.
```
Comment (codificat) [inline kubevirt/step2.md]: Is this what `virtctl start` does or there is more to it?

Note that this is an advanced approach - you can try if you are not familiar with `kubectl edit`

File: kubevirt/step2.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+Once the VM is running you can inspect it's instance:
```
Comment (codificat) [inline kubevirt/step2.md]: Grammar: "its".

But what is "its instance"?

File: kubevirt/step2.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+```
+$ kubectl get vmis
+$ kubectl get vmis -o yaml testvm
+```
+
+`kubectl get vmis`{{execute}}
+
+Once it's ready, above comand will print something like:
```
Comment (codificat) [inline kubevirt/step2.md]: Grammar: "the command above"

File: kubevirt/step2.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+~~~
+master $ kubectl get vmis
+NAME      AGE       PHASE     IP           NODENAME
+testvm    1m        Running   10.32.0.11   master
+~~~
+
+#### Accessing VMs (serial console & vnc)
+
+Now that a VM is running you can access it's console:
```
Comment (codificat) [inline kubevirt/step2.md]: Grammar: "its console"

File: kubevirt/step2.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+```
+# Connect to the serial console
+# NOTE: YOU WILL NOT BE ABLE TO ESCAPE THE CONSOLE ON KATACODA
```
Comment (codificat) [inline kubevirt/step2.md]: I could exit just fine (Chrome on Linux)... is this comment specific to some OS/browser setups?

It sounds risky to put a command to run that you can't escape from...

Also, people might not be familiar with what `^]` means, it's worth explaining how to get out of the console.

File: kubevirt/step2.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+$ ./virtctl console testvm
+```
+
+See above **NOTE** before executing:
```
Comment (codificat) [inline kubevirt/step2.md]: Then better get the NOTE out (not a comment in the command but actual text)

File: kubevirt/step2.md [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+
+`./virtctl console testvm`{{execute}}
+
+```
+# Connect to the graphical display
```
Comment (codificat) [inline kubevirt/step2.md]: As this can't be actually executed within Katacoda it's probably better to just explain the existence of `virtctl vnc` and how/where you can run it, and why you can't run it here.

File: kubevirt/step2.md [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+# It Requires remote-viewer from the virt-viewer package.
+# This obviously does not work on katacoda
+$ ./virtctl vnc testvm
+```
+
+If you've opened the console, you can click on the "+" close to 'Terminal' to start a new shell there and be able to continue with the following steps.
```
Comment (codificat) [inline kubevirt/step2.md]: "If you opened the VM's console and you can not get out of it you can click..."

Maybe this comment should be closer to the `virtctl console` command...

**Review by codificat — CHANGES_REQUESTED** [review]
A few comments with the general idea that I believe the content needs to be expanded a bit more, considering that the audience are people who are completely new to KubeVirt: it might be the very first contact they have with the technology. They might not even have a Kubernetes background - or on the contrary they might not have virtualization background.

Also a few spelling and grammar suggestions.

File: kubevirt/step1.md (hunk 4 above)
Comment (iranzo) [reply kubevirt/step1.md]: typo fixed.

about the contents we're reusing the materials we had for the minikube part (https://kubevirt.io/quickstart_minikube/) 

Would it be good to have a 'FAQ' like in metal3 site? so that those terms are explained at our website directly and users can check them without having to 'detail' in each other document?

File: kubevirt/step1.md (hunk 5 above)
Comment (iranzo) [reply kubevirt/step1.md]: We had it with JQ, and moved it to this approach to use exactly the same snippet we've at the website (and what our prow CI tests) (in fact my PR on https://github.com/kubevirt/kubevirt.github.io/pull/243 was using jq)

Whatever choice is good, but it should be 'standard' on all the docs we use

**Comment by iranzo** [comment]
Uploaded new patchset with changes requested

File: kubevirt/step1.md (hunk 5 above)
Comment (codificat) [reply kubevirt/step1.md]: I kind of disagree that the command must be the same as in other places. The reason `jq` isn't ideal in the website is because it's not as widely available as sed and awk, but the Katacoda environment does have `jq` available and it makes things simpler.

I'd value simplicity over consistency with other labs/demos, but it's up to you.

In any case it might not hurt to briefly explain what this does, again with the goal of helping people understand the contents of the lab.

File: kubevirt/step1.md (hunk 4 above)
Comment (codificat) [reply kubevirt/step1.md]: It's probably worth improving the quickstart in the website, and a FAQ might be a nice addition too, but let's handle that in the website repo.

But regardless of that, I still think it's 6 words is a bit too short for an explanation of what we're doing here.

File: kubevirt/step1.md (hunk 5 above)
Comment (iranzo) [reply kubevirt/step1.md]: thanks, let me then go back to jq and put some more indications about each steps, goals, etc

File: kubevirt/step1.md [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+
+Before we can start, we need to wait for the Kubernetes cluster to be ready (command prompt will appear to accept commands).
+
+#### Deploy KubeVirt
+
+Deploy KubeVirt operator[^1] using latest KubeVirt version (we query GH api to get latest release available).
+
+[^1]: An Operator is a method for packaging, deploying and managing Kubernetes applications, read more at [CoreOS Operators](https://coreos.com/operators/)
+
+`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases/latest|jq '.tag_name'|tr -d '"')
```
Comment (codificat) [inline kubevirt/step1.md]: Even slightly simpler with `jq`: no need to post-process wit `tr`, just use `-r`:

`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases/latest | jq 
 -r .tag_name)`

**Review by codificat — CHANGES_REQUESTED** [review]
I still see several of the comments in the previous review not addressed.

File: kubevirt/intro.md (hunk 2 above)
Comment (iranzo) [reply kubevirt/intro.md]: expanded to:

This scenario demonstrates how KubeVirt can be used to start a virtual machine

KubeVirt is a virtualization management add-on for Kubernetes.
It allows a user to run virtual machines on Kubernetes just like pods.
Because KubeVirt is integrating with Kubernetes API and resource model, the
same set of tools can be used to manage and monitor virtual machines.
 

In the current pr, is this not enough?

File: kubevirt/step1.md (hunk 4 above)
Comment (iranzo) [reply kubevirt/step1.md]: Operator was expanded to:

[^1]: An Operator is a method for packaging, deploying and managing Kubernetes applications, read more at [CoreOS Operators](https://coreos.com/operators/)


Is this not enough?

File: kubevirt/step1.md (hunk 20 above)
Comment (iranzo) [reply kubevirt/step1.md]: updated to it

**Review by codificat — CHANGES_REQUESTED** [review]
Again there are still missing comments to address.

Github's interface might be hiding them, e.g. now I see this text in the PR:

> 9 hidden conversations
>￼ Load more…

Please click the "load more" and review these.

File: kubevirt/step2.md (hunk 12 above)
Comment (iranzo) [reply kubevirt/step2.md]: virtctl does the same as using kubectl edit or even 'patch' the yaml from the command line 'alternatively' should let user be aware that there's more than 'one way to do it'

File: kubevirt/step2.md (hunk 14 above)
Comment (iranzo) [reply kubevirt/step2.md]: done

File: kubevirt/step2.md (hunk 13 above)
Comment (iranzo) [reply kubevirt/step2.md]: added link to documentation

File: kubevirt/step2.md (hunk 11 above)
Comment (iranzo) [reply kubevirt/step2.md]: it was as 'can' as you can use several methods todo it, changed to should

We're trying to start a vm, what do you mean with 'cases' ?

**Comment by iranzo** [comment]
> Again there are still missing comments to address.
> 
> Github's interface might be hiding them, e.g. now I see this text in the PR:
> 
> > 9 hidden conversations
> > ￼ Load more…
> 
> Please click the "load more" and review these.

Done, thanks !

File: kubevirt/intro.md (hunk 2 above)
Comment (ptrnull) [reply kubevirt/intro.md]: A user that gets into the katacoda scenario may not know anything about KubeVirt, we should introduce it in a first sentence, like:

> KubeVirt is a virtual machine management add-on for Kubernetes. The aim is to provide a common ground for virtualization solutions on top of Kubernetes.
And then continue with the: This scenario ...

Last paragraph I'll rephrase it to:

> KubeVirt allows a user to run virtual machines on Kubernetes just like pods. Because KubeVirt is integrating with Kubernetes API and resource model, the same set of tools can be used to manage and monitor virtual machines.

Comment (iranzo) [reply kubevirt/intro.md]: done in next pr

File: kubevirt/step1.md (hunk 4 above)
Comment (ptrnull) [reply kubevirt/step1.md]: My suggestion for [^1] is:

>  [^1] An Operator is a method of packaging, deploying and managing a Kubernetes application. A Kubernetes application is an application that is both deployed on Kubernetes and managed using the Kubernetes APIs and kubectl tooling. To be able to make the most of Kubernetes, you need a set of cohesive APIs to extend in order to service and manage your applications that run on Kubernetes. You can think of Operators as the runtime that manages this type of application on Kubernetes. If you want to learn more about Operators you can check the CoreOS Operators website: https://coreos.com/operators/

File: kubevirt/step1.md [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+Deploy KubeVirt operator[^1] using latest KubeVirt version (we query GH api to get latest release available).
+
+[^1]: An Operator is a method for packaging, deploying and managing Kubernetes applications, read more at [CoreOS Operators](https://coreos.com/operators/)
+
+`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases/latest | jq -r .tag_name)
+echo $KUBEVIRT_VERSION`{{execute}}
+
+`kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-operator.yaml`{{execute}}
+
+Provide some initial configuration to enable 'nested' virtualization in our environment. To do so, the following command enables 'emulation' to run the VM's as our demo environment is using 'emulated' virtualization. (In our environment at Katacoda, we're running inside Virtual Machines, so in order to execute a '2nd' layer of VM inside the VM we need to emulate it. In production, when using over baremetal this will not be required.)
```
Comment (ptrnull) [inline kubevirt/step1.md]: I don't understand the first sentence. I'll rephrase to:

> We need to enable 'nested' virtualization in our demo environment. To do so, we need to run the following command to run the VM's using 'emulated' virtualization. (It is a requirement of the environment in Katacoda, we're running kubernetes inside Virtual Machines, so in order to execute a '2nd' layer of VM inside the VM we need to emulate it. In production, when using over baremetal this will not be required.)

File: kubevirt/step2.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
+#### Deploy a VM
+
+Start off by creating a virtual machine:
+
+`kubectl apply -f https://raw.githubusercontent.com/kubevirt/demo/master/manifests/vm.yaml`{{execute}}
+
+Thye command above applies a `yaml` definition of a virtual machine and applies it to our current Kubernetes environment (defining vm name, resources required (disk, cpu, memory), etc).
```
Comment (ptrnull) [inline kubevirt/step2.md]: Thye should be The.
I'll write it like:

> The command above applies a `yaml` definition of a virtual machine into our current Kubernetes environment , defining the vm name, the resources required (disk, cpu, memory), etc. You can take a look to the [vm.yaml file](https://raw.githubusercontent.com/kubevirt/demo/master/manifests/vm.yaml`) if you have interest in knowing more about a virtual machine definition.

File: kubevirt/step2.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+
+This `vm` creation in the same way as we would create any other typical Kubernetes resource is what KubeVirt has enabled in our environment, making the vm's behave in a more 'Kubernetes' way.
```
Comment (ptrnull) [inline kubevirt/step2.md]: I'll rephrase to:

> We are creating a Virtual Machine in the same way as we would create any other Kubernetes resource thanks to what KubeVirt has enabled in our environment. Now we have got a Virtual Machine as a Kubernetes resource.

File: kubevirt/step2.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+
+After the vm resource has been created, you can manage/check VMs standard 'kubectl' commands:
```
Comment (ptrnull) [inline kubevirt/step2.md]: After the vm resource has been created, you can manage/check VMs standard 'kubectl' commands:

>  After the VM resource has been created, you can manage the VMs with standard 'kubectl' commands:

File: kubevirt/index.json [hunk 25]
```diff
@@ -0,0 +1,32 @@
+{
+  "pathwayTitle": "KubeVirt",
+  "title": "KubeVirt Demo",
+  "description": "Deploy and use KubeVirt in a hosted sandboxed interactive environment",
```
Comment (ptrnull) [inline kubevirt/index.json]: We can change the description to:
KubeVirt is a virtual machine management add-on for Kubernetes. Deploy it and play with it in a hosted sandboxed interactive environment.

**Comment by codificat** [comment]
One additional comment about file placement: this is the first scenario but we hope there will be more to come, and with that in mind it might be better to use a different, more specific, directory name. Something like `kubevirt-101` or `first-steps-with-kubevirt` instead of just `kubevirt`.

**Comment by ptrnull** [comment]
/lgtm
From my part, everything is all right, I also tested the scenario a couple of times and everything worked like a charm. Even the extra info helps the most beginner user to understand what is going on.

File: kubevirt-101/intro.md [hunk 26]
```diff
@@ -0,0 +1,5 @@
+KubeVirt is a virtual machine management add-on for Kubernetes. The aim is to provide a common ground for virtualization solutions on top of Kubernetes.
+
+This scenario demonstrates how KubeVirt can be used to start a virtual machine
```
Comment (codificat) [inline kubevirt-101/intro.md]: The sentence looks incomplete maybe? What about finishing it with something like:

> on a Kubernetes cluster using standard Kubernetes mechanisms.

At the very least it needs a closing period.

Comment (codificat) [inline kubevirt-101/intro.md]: The sentence looks a bit incomplete maybe? What about finishing it with something like:

> on a Kubernetes cluster using standard Kubernetes mechanisms.

At the very least, it needs a closing period.

File: kubevirt-101/intro.md [hunk 27]
```diff
@@ -0,0 +1,5 @@
...
+
+KubeVirt allows a user to run virtual machines on Kubernetes just like pods. Because KubeVirt is integrating with Kubernetes API and resource model, the same set of tools can be used to manage and monitor virtual machines.
```
Comment (codificat) [inline kubevirt-101/intro.md]: s/is integrating with/integrates with the/

File: kubevirt-101/step1.md [hunk 28]
```diff
@@ -0,0 +1,53 @@
+### Wait for the Kubernetes cluster to be ready
+
+Before we can start, we need to wait for the Kubernetes cluster to be ready (command prompt will appear to accept commands).
```
Comment (codificat) [inline kubevirt-101/step1.md]: > (a command prompt will appear once it's ready)

File: kubevirt-101/step1.md [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+
+#### Deploy KubeVirt
+
+Deploy KubeVirt operator[^1] using latest KubeVirt version.
```
Comment (codificat) [inline kubevirt-101/step1.md]: > Deploy *the* KubeVirt operator using *the* latest KubeVirt version.

File: kubevirt-101/step1.md [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+
+[^1] An Operator is a method of packaging, deploying and managing a Kubernetes application. A Kubernetes application is an application that is both deployed on Kubernetes and managed using the Kubernetes APIs and kubectl tooling. To be able to make the most of Kubernetes, you need a set of cohesive APIs to extend in order to service and manage your applications that run on Kubernetes. You can think of Operators as the runtime that manages this type of application on Kubernetes. If you want to learn more about Operators you can check the CoreOS Operators website: <https://coreos.com/operators/>
```
Comment (codificat) [inline kubevirt-101/step1.md]: I'd remove this sentence: "To be able to make the most [...] run on Kubernetes"

File: kubevirt-101/step1.md [hunk 31]
```diff
@@ -0,0 +1,53 @@
...
+
+We query GitHub api to get the latest release available (click on the text to autoexecute the commands on the console):
```
Comment (codificat) [inline kubevirt-101/step1.md]: > We query *GitHub's API* [...] latest *available release*

File: kubevirt-101/step1.md [hunk 32]
```diff
@@ -0,0 +1,53 @@
...
+
+`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases/latest | jq -r .tag_name)
+echo $KUBEVIRT_VERSION`{{execute}}
+
+Run the following command to deploy KubeVirt Operator:
+
+`kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-operator.yaml`{{execute}}
+
+We need to enable 'nested' virtualization in our demo environment. To do so, we need to run the following command to run the VM's using 'emulated' virtualization. (It is a requriment of the environment in Katacoda, we're running kubernetes inside Virtual Machines, so in order to execute a '2nd' layer of VM inside the VM we need to emulate it. In production, when using over baremetal this will not be required.)
```
Comment (codificat) [inline kubevirt-101/step1.md]: I believe the paragraph can be made a bit more clear and concise...

Technically we are not enabling "nested" virtualization, right? So I woud not mention the term "nested".

Also it's *Kubernetes* (capital K) and *bare metal* (two words).

File: kubevirt-101/step1.md [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
+Deploy KubeVirt operator[^1] using latest KubeVirt version.
+
+[^1] An Operator is a method of packaging, deploying and managing a Kubernetes application. A Kubernetes application is an application that is both deployed on Kubernetes and managed using the Kubernetes APIs and kubectl tooling. To be able to make the most of Kubernetes, you need a set of cohesive APIs to extend in order to service and manage your applications that run on Kubernetes. You can think of Operators as the runtime that manages this type of application on Kubernetes. If you want to learn more about Operators you can check the CoreOS Operators website: <https://coreos.com/operators/>
+
+We query GitHub api to get the latest release available (click on the text to autoexecute the commands on the console):
+
+`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases/latest | jq -r .tag_name)
+echo $KUBEVIRT_VERSION`{{execute}}
+
+Run the following command to deploy KubeVirt Operator:
```
Comment (codificat) [inline kubevirt-101/step1.md]: > deploy *the* KubeVirt Operator

File: kubevirt-101/step1.md (hunk 32 above)
Comment (codificat) [reply kubevirt-101/step1.md]: Maybe something along these lines:

> This demo environment already runs within a virtualized environment, and in order to be able to run VMs here we need to pre-configure KubeVirt so it uses software-emulated virtualization instead of trying to use real hardware virtualization.

File: kubevirt-101/step1.md [hunk 34]
```diff
@@ -0,0 +1,53 @@
...
+virt-controller-6849d45bcc-88zd4   1/1       Running   0          3m
+virt-controller-6849d45bcc-cmfzk   1/1       Running   0          3m
+virt-handler-fvsqw                 1/1       Running   0          3m
+virt-operator-5649f67475-gmphg     1/1       Running   0          4m
+virt-operator-5649f67475-sw78k     1/1       Running   0          4m
+~~~
+
+#### Install Virtctl
+
+`virtctl` is a client utility to provide some more convenient ways to interact with the VM (start/stop/console, etc):
```
Comment (codificat) [inline kubevirt-101/step1.md]: > client utility *that helps interact with VMs*

File: kubevirt-101/step2.md [hunk 35]
```diff
@@ -0,0 +1,80 @@
+#### Deploy a VM
+
+The command below applies a yaml definition of a virtual machine into our current Kubernetes environment, defining the vm name, the resources required (disk, cpu, memory), etc. You can take a look to the [vm.yaml](https://raw.githubusercontent.com/kubevirt/demo/master/manifests/vm.yaml) file if you have interest in knowing more about a virtual machine definition:
```
Comment (codificat) [inline kubevirt-101/step2.md]: Acronyms in capitals:

> [...] a YAML definition [...] VM name [...] CPU

Also "take a look at".

File: kubevirt-101/step2.md [hunk 36]
```diff
@@ -0,0 +1,80 @@
...
+
+`kubectl apply -f https://raw.githubusercontent.com/kubevirt/demo/master/manifests/vm.yaml`{{execute}}
+
+We are creating a Virtual Machine in the same way as we would create any other Kubernetes resource thanks to what KubeVirt has enabled in our environment. Now we have got a Virtual Machine as a Kubernetes resource.
```
Comment (codificat) [inline kubevirt-101/step2.md]: > Now we have a Virtual

(without "got")

File: kubevirt-101/step2.md [hunk 37]
```diff
@@ -0,0 +1,80 @@
...
+
+```
+$ kubectl get vms
+$ kubectl get vms -o yaml testvm
+```
+
+Check VM's defined (using commands above)
+
+`kubectl get vms`{{execute}}
+
```
Comment (codificat) [inline kubevirt-101/step2.md:19]: Maybe add an additional comment here:

> Notice from the output of the command that the VM is not running yet.

File: kubevirt-101/intro.md (hunk 26 above)
Comment (ptrnull) [reply kubevirt-101/intro.md]: We are already saying so in the paragraph above.  The closing period works for me.

File: kubevirt-101/step2.md [hunk 38]
```diff
@@ -0,0 +1,80 @@
...
+`./virtctl start testvm`{{execute}}
+
+Now, you can check again the vm status:
+
+`kubectl get vms`{{execute}}
+
+Alternatively you can use `kubectl edit vm testvm` to set `.spec.running: true`.
+
+Once the VM is running you can inspect its status as [instance](https://kubevirt.io/user-guide/docs/latest/creating-virtual-machines/intro.html) :
```
Comment (codificat) [inline kubevirt-101/step2.md]: I believe this is not clear enough, thinking in someone who sees KubeVirt for the first time. I would expand this a bit more:

> A `VirtualMachine` resource contains a VM's definition and status. An instance of a running VM has an additional associated resource: a `VirtualMachineInstance`.
>
> Once the VM is running you can inspect its instance's status:

File: kubevirt-101/step2.md [hunk 39]
```diff
@@ -0,0 +1,80 @@
...
+master $ kubectl get vmis
+NAME      AGE       PHASE     IP           NODENAME
+testvm    1m        Running   10.32.0.11   master
+~~~
+
+#### Accessing VMs (serial console & vnc)
+
+Now that a VM is running you can access its console:
+
+**NOTE** : YOU WILL NOT BE ABLE TO ESCAPE THE CONSOLE ON KATACODA.
```
Comment (codificat) [inline kubevirt-101/step2.md]: I just tried again and I can exit from the console just fine.

In which environment are you having that problem? My test was with Chrome on Linux.

Maybe mention as a WARNING that "in some browser environments you will not be able to..."

File: kubevirt-101/step2.md [hunk 40]
```diff
@@ -0,0 +1,80 @@
...
+
+`^]` means: press CTRL + "]" keys to escape the console
```
Comment (codificat) [inline kubevirt-101/step2.md]: Add **NOTE** at the beginning

File: kubevirt-101/step2.md [hunk 41]
```diff
@@ -0,0 +1,80 @@
...
+~~~sh
+# Connect to the serial console
+$ ./virtctl console testvm
+~~~
+
+If you've opened the console **within** Katacoda, you can click on the `+` close to 'Terminal' to start a new shell there and be able to continue with the following steps in the shutdown and cleanup section.
+
+**NOTE** This doesn't work on Katacoda because of the environment setup and browser access
+
+`./virtctl console testvm`
```
Comment (codificat) [inline kubevirt-101/step2.md]: What doesn't work?

This is again the same `virtctl console` command as above, it looks like a duplicate...

File: kubevirt-101/step2.md [hunk 42]
```diff
@@ -0,0 +1,80 @@
...
+
+~~~sh
```
Comment (codificat) [inline kubevirt-101/step2.md]: Ok so I guess that the note above about "this doesn't work" was talking about VNC access.

IMHO then just explain that in environments where VNC client access is available it's possible to access the graphical console of VMs using `virtctl vnc` - and link to the relevant doc instead of writing the example command (that can't be executed in this environment).

**Review by codificat — CHANGES_REQUESTED** [review]
Looking better thanks!

Just a few additional comments and suggestions.
